src/main/java/org/prebid/server/bidder/mediasquare/MediasquareBidder.java

```diff
@@ -0,0 +1,318 @@
+
+        return MediasquareCode.builder()
+                .adUnit(imp.getTagid())
+                .auctionId(bidRequest.getId())
+                .bidId(imp.getId())
+                .code(extImp.getCode())
+                .owner(extImp.getOwner())
+                .mediaTypes(mediaTypes)
+                .floor(floors)
+                .build();
+    }
+
+    private static MediasquareMediaTypes makeMediaTypes(Imp imp) {
+        final Video video = imp.getVideo();
+        final Banner banner = imp.getBanner();
+        final Native xNative = imp.getXNative();
+
+        if (video == null && banner == null && xNative == null) {
+            return null;
+        }
+
+        return MediasquareMediaTypes.builder()
+                .banner(makeBanner(banner))
+                .video(video)
+                .nativeRequest(xNative != null ? xNative.getRequest() : null)
+                .build();
+    }
+
+    private static MediasquareBanner makeBanner(Banner banner) {
+        if (banner == null) {
+            return null;
+        }
+
+        final List<List<Integer>> sizes = new ArrayList<>();
+        if (CollectionUtils.isNotEmpty(banner.getFormat())) {
+            for (Format format : banner.getFormat()) {
+                if (format != null) {
+                    sizes.add(List.of(format.getW(), format.getH()));
+                }
+            }
+        } else if (banner.getW() != null && banner.getH() != null) {
+            sizes.add(List.of(banner.getW(), banner.getH()));
+        }
+
+        return sizes.isEmpty() ? null : MediasquareBanner.of(sizes);
+    }
+
+    private static Map<String, MediasquareFloor> makeFloors(Imp imp) {
+        final Map<String, MediasquareFloor> floors = new HashMap<>();
+        final MediasquareFloor floor = MediasquareFloor.of(imp.getBidfloor(), imp.getBidfloorcur());
+
+        final Video video = imp.getVideo();
+        final Banner banner = imp.getBanner();
+        final Native xNative = imp.getXNative();
+
+        if (video != null) {
+            if (video.getW() != null && video.getH() != null) {
+                final String videoSize = SIZE_FORMAT.formatted(video.getW(), video.getH());
+                floors.put(videoSize, floor);
+            }
+            floors.put("*", floor);
+        }
+
+        if (banner != null) {
+            if (CollectionUtils.isNotEmpty(banner.getFormat())) {
+                for (Format format : banner.getFormat()) {
+                    if (format != null) {
+                        floors.put(SIZE_FORMAT.formatted(format.getW(), format.getH()), floor);
+                    }
+                }
+            } else if (banner.getW() != null && banner.getH() != null) {
+                floors.put(SIZE_FORMAT.formatted(banner.getW(), banner.getH()), floor);
+            }
+        }
+
+        if (xNative != null) {
+            floors.put("*", floor);
+        }
+
+        return MapUtils.isNotEmpty(floors) ? floors : null;
+    }
```

Comment:
Change logic a bit, so it will construct floors map from `MediasquareMediaTypes`